design-documents/graph-ql/schema-development.md

```diff
@@ -0,0 +1,85 @@
+# [GraphQL][] Schema Development
+
+Before developing a [GraphQL schema](https://graphql.org/learn/schema/), we need to consider the following:
+
+- [GraphQL Ecosystem](#graphql-ecosystem)
+- [GraphQL Server](#graphql-server)
+- [Code-First vs. SDL](#code-first-vs-sdl)
+- [Code-First Frameworks](#code-first-frameworks)
+   - [Nexus](#nexus)
+
+## GraphQL Ecosystem
+
+[GraphQL][] is [a specification](https://graphql.github.io/graphql-spec/) for which a good handful of implementations are written across various languages (e.g., [Rust](https://github.com/graphql-rust), [PHP](https://github.com/webonyx/graphql-php), [Python](https://graphene-python.org/), [.NET](https://github.com/graphql-dotnet/graphql-dotnet) and [JavaScript](https://www.npmjs.com/package/graphql)). There is no wrong choice between these implementations; however, when considering the ecosystem, there is a clear winner.
+
+In 2015, the [GraphQL][] spec was [first published](https://graphql.github.io/graphql-spec/July2015/) along with the reference implementation in JavaScript. It should be no surprise then, that today, JavaScript holds the strongest ecosystem of [GraphQL][] libraries, with [4,833 related npm packages](https://www.npmjs.com/search?q=graphql) at the time of this writing (10/2019). As such, the JavaScript ecosystem is a solid and safe choice, moving forward.
+
+## GraphQL Server
+
+Within the JavaScript ecosystem, there are [a few server choices available](https://www.npmtrends.com/apollo-server-vs-koa-graphql-vs-prisma), but the most popular and mature choice is [Apollo Server][]. Optionally, [`graphql-yoga`](https://www.npmjs.com/search?q=graphql-yoga) uses [Apollo Server][] under the hood.
```

Comment:
- graphql-yoga should be deprecated, it's based on Apollo Server 1.0. The things it does are already in Apollo Server 2.0
- Prisma should be considered as an ORM with deep integrations with the frontend API, has nothing to do with the actual server (especially with Prisma 2.0)

Reply:
See https://github.com/magento/architecture/pull/332/commits/f3fb8d0bc26d11380d4855c0b895ebcc0151dcce